### ISSUE TITLE
Rename TrackType to Library

### ISSUE THREAD

**Issue by jooola** [libretime/libretime#1922]
I propose to rename the track types to libraries, but in the api only.

The way we currently understand track types is that they are optional and could help one to sort and filter files using this criteria.

By renaming this field to library, the field will become the ancestor of files and should be required for any upload.

[![](https://mermaid.ink/svg/pako:eNo1zrESgjAMBuBX6WWGQUcGF9FJJ9koQ6SB9oSWK6l3HPDuFsVMufzfJZmhdoogg9bjoEWRSytiNYfyajqqRJqexHI3rWah8U2L4LLwWL8ET0OMf7orb-bp0U-7PzvLaOy4iOb43TNWG4QEevI9GhXvzdtEAmvqSUIWW0UNho4lSLtGGgaFTBdl2HnIGuxGSgADu8dka8jYB_qj3GB8v9_V-gGp3Eb3)](https://mermaid.live/edit#pako:eNo1zrESgjAMBuBX6WWGQUcGF9FJJ9koQ6SB9oSWK6l3HPDuFsVMufzfJZmhdoogg9bjoEWRSytiNYfyajqqRJqexHI3rWah8U2L4LLwWL8ET0OMf7orb-bp0U-7PzvLaOy4iOb43TNWG4QEevI9GhXvzdtEAmvqSUIWW0UNho4lSLtGGgaFTBdl2HnIGuxGSgADu8dka8jYB_qj3GB8v9_V-gGp3Eb3)

With this proposal, I would like change the file storage path to a similar schema: `{config.storage.path}/files/{file.library.name}/{file.owner.id}/{file.some_path_we_think_right}`

This should allow to run specific actions to specific libraries, and have options on those libraries. For example, we might want to scan a music library differently than a show library, or a jingle library. Or have some stricter analysis settings on some libraries. Or use some libraries to have a user defined file structure, and import should happen during library scan. There are many ideas.

**Comment by paddatrapper** [comment]
Yes, this is a very good idea. Different libraries definitely have different handling criteria
